chessboard/olegbet/chessboard.go

```diff
@@ -0,0 +1,54 @@
+package chessboard
+
+// Declare a type named Rank which stores if a square is occupied by a piece - this will be a slice of bools
+
+type Rank []bool
+
+// Declare a type named Chessboard contains a map of eight Ranks, accessed with values from "A" to "H"
+
+type Chessboard map[string]Rank
+
+// CountInRank returns how many squares are occupied in the chessboard,
+// within the given rank
+func CountInRank(cb Chessboard, rank string) (count int) {
+	r := cb[rank]
+	for _, v := range r {
+		if v {
+			count++
+		}
+	}
+	return
+}
+
+// CountInFile returns how many squares are occupied in the chessboard,
+// within the given file
+func CountInFile(cb Chessboard, file int) (count int) {
+	if file >= 0 && file <= len(cb["A"]) {
+		for _, v := range cb {
+			if v[file-1] {
+				count++
+			}
+		}
+	}
+	return
+}
+
+// CountAll should count how many squares are present in the chessboard
+func CountAll(cb Chessboard) (count int) {
+	for _, v := range cb {
```

Comment:
If your ranks have a given size (8), this function would become something like:
```go
    return len(cb) * 8
```
Although to be honest we could do the same with the number of ranks and return an int... But you get my point.